Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -84,9 +112,13 @@ def fixed_name(file, bad_name)
     elsif bad_name.start_with? CELLAR_PLACEHOLDER
       bad_name.sub(CELLAR_PLACEHOLDER, HOMEBREW_CELLAR)
     elsif (file.dylib? || file.mach_o_bundle?) && (file.parent + bad_name).exist?
-      "@loader_path/#{bad_name}"
+      file.parent/bad_name
     elsif file.mach_o_executable? && (lib + bad_name).exist?
       "#{lib}/#{bad_name}"
+    elsif bad_name.start_with?("@rpath") && ENV["HOMEBREW_RELOCATE_METAVARS"]
+      expand_rpath file, bad_name
+    elsif bad_name.start_with?("@loader_path") && ENV["HOMEBREW_RELOCATE_METAVARS"]
+      expand_loader_path file, bad_name
```

Comment:
Is there an advantage to rewriting `@loader_path`, or is it simpler to leave it as is?

Reply:
Oh, that was as a follow up to https://github.com/Homebrew/homebrew-core/issues/36685 -- I don't think there's an advantage per se (we're just doing AOT what `dyld` would do at load-time). Perhaps I misunderstood what you were asking for in that issue?

Reply:
That's exactly what I said, but I spoke too quickly. After re-reading https://github.com/Homebrew/homebrew-core/issues/36685, it's not clear to me what exactly the root-problem is. Sorry for sending you astray.

This change would make some kegs that were previously `cellar :any_skip_relocatable` because they use `@loader_path` to find their libraries would require relocation and become `cellar :any` after this change. For this reason, I don't think this change is desirable.

Reply:
What does AOT stand for?

Reply:
No problem! And sorry -- AOT is "ahead of time." But you're right -- it just adds a relocation case where none is necessary.